daemon/main.go

```diff
@@ -810,6 +811,47 @@ func runDaemon() {
 	log.Info("Launching node monitor daemon")
 	go d.nodeMonitor.Run(path.Join(defaults.RuntimePath, defaults.EventsPipe), bpf.GetMapRoot())
 
+	eventsCh, err := workloads.EnableEventListener()
+	if err != nil {
+		log.WithError(err).Fatal("Error while enabling docker event watcher")
+	} else {
+		d.workloadsEventsCh = eventsCh
+	}
+
+	if err := d.EnableK8sWatcher(5 * time.Minute); err != nil {
```

Comment:
I am wary of changing the ordering of operations in the bootstrapping of the daemon. For instance, how does putting restore after the K8s watcher / workloads event listener affect the starting of the agent? This is unclear at this time. I could maintain the same order, and communicate via channels to make things more concurrent to speed up said bootstrapping. It also behooves us to write why certain operations are done in certain orders in said bootstrapping code, because this has bitten us before.